### PR TITLE
Readme update

### PR DESCRIPTION
**Pull request by mbasha-dell**
# Description
This PR is to update the readme to show more accurate description & function of the repo. Also removed unused links and sections. 

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/1422 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?

- [x] Unit testing
- [ ] Integration testing

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -45,7 +45,7 @@ vendor:	##generate the vendor directory
 	go mod vendor
 
 .PHONY: build
-build:	##build project
+build: ##build project
```
Comment (lukeatdell) [inline Makefile:48]: `build` should probably execute `codegen` as a prerequisite. 
There's no instruction in the README to execute `codegen` and it's possible the generated code could go stale during development. I think adding it here covers both issues.

Comment (mbasha-dell) [reply Makefile:48]: I agree with the idea, but in practice for some reason adding `codegen` to the build line was resulting in `make podman` failing, although `make build` worked just fine.

**Review by lukeatdell — APPROVED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -35,6 +35,7 @@ COPY overrides.mk overrides.mk
 COPY Makefile Makefile
 COPY cmd/main.go cmd/main.go
 COPY pkg/ pkg/
+COPY .mockery.yaml .mockery.yaml
```
Comment (donatwork) [inline Dockerfile:38]: Why did you need to copy this to the Dockerfile. Isn't mockery a test related item? This should not be in the production image.

File: pkg/config/config.go [hunk 3]
```diff
@@ -26,7 +26,7 @@ import (
 	"gopkg.in/yaml.v3"
 )
 
-//go:generate go run github.com/atombender/go-jsonschema/cmd/gojsonschema@main --package=config --output=config.gen.go config.schema.json --extra-imports
+//go:generate go run github.com/atombender/go-jsonschema/cmd/gojsonschema@v0.13.1 --package=config --output=config.gen.go config.schema.json --extra-imports
```
Comment (donatwork) [inline pkg/config/config.go:29]: What was the reason for changing this?

File: Dockerfile (hunk 2 above)
Comment (mbasha-dell) [reply Dockerfile:38]: This is a dependency for `codegen `to work. `make podman` will fail unless go's code generation is provided this file. Refer to the following PR to find the failed build action - https://github.com/dell/cosi/pull/202/commits/85f6d6826f95655a0eb7c2ebd795655d6e74cac2

File: pkg/config/config.go (hunk 3 above)
Comment (mbasha-dell) [reply pkg/config/config.go:29]: image won't build when referencing the main branch

File: Dockerfile (hunk 2 above)
Comment (mbasha-dell) [reply Dockerfile:38]: Addressed in the following, per 🥇 @lukeatdell 🥇help 
https://github.com/dell/cosi/commit/8ddc467603639c2bb98987df26007c9ea88de11a

**Review by lukeatdell — DISMISSED** [review]

File: Dockerfile (hunk 2 above)
Comment (lukeatdell) [reply Dockerfile:38]: Oh don't forget to remove this COPY now that it's not needed

Comment (mbasha-dell) [reply Dockerfile]: Ah shoot! Addressed again here, thanks guys:
https://github.com/dell/cosi/pull/202/commits/90457a15d20b3e42dc139f63da1dcf8827c08dd8

**Review by lukeatdell — DISMISSED** [review]

**Review by donatwork — DISMISSED** [review]

**Comment by donatwork** [comment]
Approval pending an attempt at the integration or e2e tests.

File: .github/workflows/development.yaml [hunk 4]
```diff
@@ -142,7 +142,7 @@ jobs:
          chmod +x ./scripts/build-ubi-micro.sh
          make build-base-image
          podman build -t docker.io/${{ env.IMAGENAME }}:${{ env.IMAGETAG }} -f ./Dockerfile --build-arg GOIMAGE=golang:latest --build-arg BASEIMAGE="localhost/cosi-ubimicro"
-         podman save docker.io/library/${{ env.IMAGENAME }}:${{ env.IMAGETAG }} -o /tmp/cosi.tar
```
Comment (donatwork) [inline .github/workflows/development.yaml:145]: Why was this change needed?

File: .github/workflows/development.yaml [hunk 5]
```diff
@@ -142,7 +142,7 @@ jobs:
...
+         podman save docker.io/${{ env.IMAGENAME }}:${{ env.IMAGETAG }} -o /tmp/cosi.tar
```
Comment (donatwork) [inline .github/workflows/development.yaml:145]: How did this ever work? Are we shipping bad code in previous releases?

Comment (mbasha-dell) [reply .github/workflows/development.yaml:145]: I'm also curious about this. I know that running the commands to build the image via the instructions in the readme would cause a failure here as it couldn't find the image when library was in the path. 

One thing of note is that we had to switch over from Docker to Podman. I am not sure if there is any image path resolution done when there is `library` within the file path, or some other Docker-> Podman related differences that would cause this, but that would be my best guess.

**Review by donatwork — APPROVED** [review]

**Review by lukeatdell — APPROVED** [review]

**Review by suryagupta4 — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (mbasha-dell) [reply Makefile]: codegen is back in so marking this as resolved

File: Makefile [hunk 6]
```diff
@@ -55,20 +59,20 @@ build:	##build project
 .PHONY: build-base-image
 build-base-image: vendor download-csm-common
 	$(eval include csm-common.mk)
-	sh ./scripts/build-ubi-micro.sh $(DEFAULT_BASEIMAGE)
+	sh ./scripts/build-ubi-micro.sh $(DEFAULT_BASEIMAGE);
 	$(eval BASEIMAGE=cosi-ubimicro:latest)
 
-.PHONY: docker
-docker: build-base-image
+.PHONY: podman
```
Comment (donatwork) [inline Makefile:65]: In hindsight. The target should have been kept as docker. These are still "Docker" images. The tool itself could have changed to podman but the target should have remained as external tools may still expect a target as docker.
